log4j-layout-template-json/src/test/java/org/apache/logging/log4j/layout/template/json/GcpLayoutTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+package org.apache.logging.log4j.layout.template.json;
+
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.DefaultConfiguration;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.logging.log4j.layout.template.json.TestHelpers.serializeUsingLayout;
+
+class GcpLayoutTest {
+
+    private static final Configuration CONFIGURATION = new DefaultConfiguration();
+
+    private static final Charset CHARSET = StandardCharsets.UTF_8;
+
+    private static final String SERVICE_NAME = "test";
+
+    private static final JsonTemplateLayout JSON_TEMPLATE_LAYOUT = JsonTemplateLayout
+            .newBuilder()
+            .setConfiguration(CONFIGURATION)
+            .setCharset(CHARSET)
+            .setEventTemplateUri("classpath:GcpLayout.json")
+            .build();
+
+    @Test
+    void test_lite_log_events() {
+        final List<LogEvent> logEvents = LogEventFixture.createLiteLogEvents(1_000);
+        test(logEvents);
+    }
+
+    @Test
+    void test_full_log_events() {
+        final List<LogEvent> logEvents = LogEventFixture.createFullLogEvents(1_000);
+        test(logEvents);
+    }
+
+    private static void test(final Collection<LogEvent> logEvents) {
+        for (final LogEvent logEvent : logEvents) {
+            test(logEvent);
+        }
+    }
+
+    private static void test(final LogEvent logEvent) {
+        final Map<String, Object> jsonTemplateLayoutMap = renderUsingJsonTemplateLayout(logEvent);
+        // for now, we just validate there are no errors
```

Comment:
@rocketraman, can we do something more meaningful here, e.g., checking a couple of fields against the `logEvent`?

Reply:
For some reason the delegated pattern resolver doesn't actually seem to be applying in the test case. For example, here is an example output:

```
{timestamp=2021-07-05T13:30:58.150Z, severity=%level, message=%m%xEx, logging.googleapis.com/sourceLocation={function=%C.%M}, logging.googleapis.com/insertId=%sn, _thread=ForkJoinPool-1-worker-18, _logger=a.B0}
```

The severity, message, and sourceLocation fields which all use the pattern resolver haven't been resolved. Any ideas?

Reply:
```
ERROR StatusLogger Unrecognized format specifier [d]
ERROR StatusLogger Unrecognized conversion specifier [d] starting at position 16 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [thread]
ERROR StatusLogger Unrecognized conversion specifier [thread] starting at position 25 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [level]
ERROR StatusLogger Unrecognized conversion specifier [level] starting at position 35 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [logger]
ERROR StatusLogger Unrecognized conversion specifier [logger] starting at position 47 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [msg]
ERROR StatusLogger Unrecognized conversion specifier [msg] starting at position 54 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [n]
ERROR StatusLogger Unrecognized conversion specifier [n] starting at position 56 in conversion pattern.

```

Reply:
These seem like a plugin mishap to me. How do you run the tests? I guess you try to run them from IDE, IntelliJ? Would you mind trying the following, please?
```bash
./mvnw clean install -DskipTests && ./mvnw -pl log4j-layout-template-json test -Dtest=GcpLayoutTest
```
If this succeeds, after `clean install`, try clicking on the _Reload All Maven Projects_ button in IntelliJ and run the test class again.

Reply:
@vy That worked, thank you. I've updated the test case (and fixed some problems the test revealed) in 81aa8dbce.

Your thoughts on these lines in the test would be useful:

1. https://github.com/apache/logging-log4j2/pull/543/commits/81aa8dbcef3b3999496dffd6ae605c6624e46fd1#diff-3498fad3fa3b9848532a5406e402db89f1ec033493bf2770f94eb0c47d5b192cR94-R96
2. https://github.com/apache/logging-log4j2/pull/543/commits/81aa8dbcef3b3999496dffd6ae605c6624e46fd1#diff-3498fad3fa3b9848532a5406e402db89f1ec033493bf2770f94eb0c47d5b192cR123-R127

Reply:
@vy While you research the additional fields stuff, just a reminder on this -^

Reply:
Checked these out and addressed them in #551.